llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1140,12 +1125,18 @@ static VPActiveLaneMaskPHIRecipe *addVPLaneMaskPhiAndUpdateExitBranch(
   // original terminator.
   VPRecipeBase *OriginalTerminator = EB->getTerminator();
   Builder.setInsertPoint(OriginalTerminator);
-  auto *InLoopIncrement =
-      Builder.createOverflowingOp(VPInstruction::CanonicalIVIncrementForPart,
-                                  {IncrementValue}, {false, false}, DL);
-  auto *ALM = Builder.createNaryOp(VPInstruction::ActiveLaneMask,
-                                   {InLoopIncrement, TripCount}, DL,
-                                   "active.lane.mask.next");
+
+  auto *InLoopIncrement = CanonicalIVIncrement;
+  // When the loop is guarded by a runtime overflow check for the loop induction
```

Comment:
I might be missing something, but if `DataAndControlFlowWithoutRuntimeCheck` there is no runtime check to make sure `iv + UFxVF` doesn't overflow.

Reply:
hi @fhahn 
 Do you mean use **-mllvm -force-tail-folding-style=data-and-control-without-rt-check** to enable **DataAndControlFlowWithoutRuntimeCheck** ? I don't find any change with the option, see https://gcc.godbolt.org/z/M878YbK8d